pandas/core/base.py

```diff
@@ -884,6 +890,21 @@ def argmin(self, axis=None):
         """
         return nanops.nanargmin(self.values)
 
+    def tolist(self):
+        """
+        return a list of the values; box to scalars
+        """
+        return list(self.__iter__())
+
+    def __iter__(self):
+        """
+        provide iteration over the values; box to scalars
+        """
+        if is_datetimelike(self):
+            return (_maybe_box_datetimelike(x) for x in self._values)
+        else:
+            return iter(self._values.tolist())
```

Comment:
For the `tolist` implementation, this seems a bit double work: values are converted to list, then iterateted over, and then again converted to list.

Reply:
fixed